src/app/features/notification/notification-rule/notification-rule.component.ts

```diff
@@ -65,24 +84,57 @@ export class NotificationRuleComponent implements OnChanges {
 
   initForm() {
     this.form = new FormGroup({
-      entityType: new FormControl(this.value?.entityType ?? ""),
+      entityType: new FormControl({
+        value: this.value?.entityType ?? "",
+        disabled: Object.keys(this.value?.conditions ?? {}).length > 0,
+      }),
       enabled: new FormControl(this.value?.enabled || false),
       // different format for form control
       channels: new FormControl(
         this.parseChannelsToOptionsArray(this.value?.channels),
       ),
-      conditions: new FormControl(this.value?.conditions ?? ""), // TODO: parse conditions format?
-      // hidden fields
+      conditions: new FormArray(
+        this.parseConditionsObjectToArray(this.value?.conditions).map(
+          (c) =>
+            new FormGroup({
+              entityTypeField: new FormControl(c.entityTypeField),
+              operator: new FormControl(c.operator),
+              condition: new FormControl(c.condition),
+            }),
+        ),
+      ),
       notificationType: new FormControl(
         this.value?.notificationType ?? "entity_change",
       ),
     });
 
+    this.updateEntityTypeControlState();
     this.form.valueChanges.subscribe((value) => this.updateValue(value));
   }
 
+  /**
+   * Disable the entityType field if there are notification conditions.
+   */
+  private updateEntityTypeControlState() {
+    this.form.get("conditions").valueChanges.subscribe(() => {
+      const conditionsLength = (this.form.get("conditions") as FormArray)
+        .length;
```

Comment:
This will crash, if parsing is not possible. Please catch this.

Reply:
Thanks for this suggestion. I have updated the code accordingly. Please check and let me know if there are any suggested changed needed.